### ISSUE TITLE
GPU Memory consumption increasing during training

### ISSUE THREAD

**Issue by 1pha** [1pha/MulitmodalERC-TensorMixerNetwork#23]
계속해서 훈련 중에 GPU Memory allocation값이 선형적으로 증가해서 하나의 epoch이 끝나지 못하고 train이 CUDA OOM으로 끝나버리는 경우가 있었다. 제일 중요한 포인트는 `pl.LightningModule`에서 `self.*_step` 메소드에서 반환되는 값들은 **뒤에서 계속 쌓이게된다**. 이 값들은 GPU에서 계속 쌓이는데 여기서 문제는 직접 저장한 값들이 그렇게 shape이 크지 않았다는 점에서 의문이 시작됐다.

결과적으로는 반환해야 하는 값 중 실제로 gradient backpropagation을 수행하지 **않아도**되는 값들이 `.detach()`를 진행해주지 않아 GPU Memory allocation이 계속해서 증가하게 된 것이었다. 다음 번에는 꼭 참조...
https://github.com/Lightning-AI/lightning/discussions/13648

<img width="380" alt="image" src="https://user-images.githubusercontent.com/62973169/224562504-a2f013fc-a17f-4857-915f-ef7676bf9f37.png">
<img width="640" alt="image" src="https://user-images.githubusercontent.com/62973169/224562523-e6f70645-8e6e-4b2a-8719-55c07be38dd6.png">
